### PR TITLE
add geo: cylinder in cube

### PR DESCRIPTION
**Pull request by Liu-xiandong**
add geo: cylinder in cube

image

### CODE REVIEW

File: paddlescience/discretize.py [hunk 1]
```diff
@@ -41,9 +41,13 @@ def sampling_discretize(pde,
                         geo,
                         time_nsteps=None,
                         space_point_size=None,
```
Comment (xingfeng01) [inline paddlescience/discretize.py:43]: space_npoints

File: paddlescience/discretize.py [hunk 2]
```diff
@@ -41,9 +41,13 @@ def sampling_discretize(pde,
...
+                        space_nsteps=None,
+                        circle_bc_size=None):
     # Geometry
-    geo_disc = geo.sampling_discretize(time_nsteps, space_point_size,
-                                       space_nsteps)
+    geo_disc = geo.sampling_discretize(
+        time_nsteps,
+        space_point_size,
+        space_nsteps,
+        circle_bc_size=circle_bc_size)
```
Comment (xingfeng01) [inline paddlescience/discretize.py:51]: delete circle_bc_size

Comment (xingfeng01) [inline paddlescience/discretize.py:51]: delete circle_bc_size

File: paddlescience/geometry/cylinder_in_cube.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    """
+    Two dimentional rectangular
+
+    Parameters:
+        space_origin: cordinate of left-bottom point of rectangular
+        space_extent: extent of rectangular
+
+    Example:
+        >>> import paddlescience as psci
+        >>> geo = psci.geometry.Rectangular(space_origin=(0.0,0.0), space_extent=(1.0,1.0))
```
Comment (xingfeng01) [inline paddlescience/geometry/cylinder_in_cube.py:35]: update comments

Comment (xingfeng01) [inline paddlescience/geometry/cylinder_in_cube.py:35]: update comments

File: paddlescience/discretize.py (hunk 2 above)
Comment (Liu-xiandong) [reply paddlescience/discretize.py]: Done

File: paddlescience/discretize.py (hunk 1 above)
Comment (Liu-xiandong) [reply paddlescience/discretize.py]: Done

File: paddlescience/discretize.py (hunk 2 above)
Comment (Liu-xiandong) [reply paddlescience/discretize.py]: Done

File: paddlescience/geometry/cylinder_in_cube.py (hunk 3 above)
Comment (Liu-xiandong) [reply paddlescience/geometry/cylinder_in_cube.py]: Done

Comment (Liu-xiandong) [reply paddlescience/geometry/cylinder_in_cube.py]: Done

**Review by xingfeng01 — APPROVED** [review]

**Pull request merged**